src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java

```diff
@@ -1245,6 +1247,7 @@ public String replaceSelection(String value, boolean collapseSelection)
    {
       Selection selection = getSession().getSelection();
       String oldValue = getSession().getTextRange(selection.getRange());
+      value = StringUtil.normalizeNewLines(value);
```

Comment:
Should this be done in the single argument `replaceSelection` rather than here? (seems like we're still open to the same problem if that function is called directly).